keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryArtifactRepository.kt

```diff
@@ -391,6 +391,12 @@ class InMemoryArtifactRepository(
     val artifactId = getId(artifact) ?: throw NoSuchArtifactException(artifact)
     val key = EnvironmentVersionsKey(artifactId, deliveryConfig, targetEnvironment)
     val statuses = statusByEnvironment.getOrPut(key, ::mutableMapOf)
+    // update all previous deploying things to skipped
+    val deployingVersions = statuses.filterValues { it == DEPLOYING }
+    deployingVersions.keys.forEach {
+      log.error("Stuck deploying version $it for artifact '${artifact.reference}' in delivery config ${deliveryConfig.name} found when deploying version $version")
```

Comment:
You mentioned Radar would alert us of this error, but I don’t think it will because it monitors exceptions, not any error logs.

Reply:
I'll figure out how to get something to alert us